main.go

```diff
@@ -5,61 +5,110 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"io/ioutil"
+	"io"
+	"log/slog"
 	"net/http"
 	"os"
+	"slices"
 	"strings"
 
-	"github.com/caarlos0/env/v6"
-	egoScaleLog "github.com/exoscale-labs/snap-o-matic/log"
-	"github.com/exoscale/egoscale"
 	flag "github.com/spf13/pflag"
-	log "gopkg.in/inconshreveable/log15.v2"
+
+	v3 "github.com/exoscale/egoscale/v3"
+	"github.com/exoscale/egoscale/v3/credentials"
 )
 
 const (
 	metadataURL = "http://169.254.169.254/latest/meta-data/vm-id"
```

Comment:
```suggestion
	metadataURL = "http://metadata.exoscale.com/latest/meta-data/vm-id"
```

Reply:
https://community.exoscale.com/documentation/compute/cloud-init/#querying-the-user-data-and-meta-data-from-the-instance